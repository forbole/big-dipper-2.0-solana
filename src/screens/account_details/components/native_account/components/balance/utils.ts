import { formatNumber } from '@utils/format_token';
import { BalanceType } from './types';

export const formatBalanceData = (data: BalanceType) => {
  const balanceChart = [
    {
      key: 'balanceNative',
      display: `${formatNumber(data.native.value, data.native.exponent)} ${data.native.displayDenom.toUpperCase()}`,
      value: data.native.value,
    },
    {
      key: 'balanceStake',
      display: `${formatNumber(data.stake.value, data.stake.exponent)} ${data.stake.displayDenom.toUpperCase()}`,
      value: data.stake.value,
    },
    {
      key: 'balanceNonce',
      display: `${formatNumber(data.nonce.value, data.nonce.exponent)} ${data.nonce.displayDenom.toUpperCase()}`,
      value: data.nonce.value,
    },
    {
      key: 'balanceVote',
      display: `${formatNumber(data.vote.value, data.vote.exponent)} ${data.vote.displayDenom.toUpperCase()}`,
      value: data.vote.value,
    },
  ];

  return balanceChart;
};
