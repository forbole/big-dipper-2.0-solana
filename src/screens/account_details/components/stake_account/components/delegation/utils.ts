import numeral from 'numeral';
import { DelegationType } from './types';

export const formatBalanceData = (data: DelegationType) => {
  const balanceChart = [
    {
      key: 'balanceActivated',
      display: `${numeral(data.activated.value).format(data.activated.format)} ${data.activated.denom.toUpperCase()}`,
      value: data.activated.value,
    },
    {
      key: 'balanceActivating',
      display: `${numeral(data.activating.value).format(data.activating.format)} ${data.activating.denom.toUpperCase()}`,
      value: data.activating.value,
    },
  ];

  return balanceChart;
};
