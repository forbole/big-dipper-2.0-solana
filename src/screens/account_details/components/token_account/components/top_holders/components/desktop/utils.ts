export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 5,
    align: 'center',
  },
  {
    key: 'holder',
    width: 65,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
];
