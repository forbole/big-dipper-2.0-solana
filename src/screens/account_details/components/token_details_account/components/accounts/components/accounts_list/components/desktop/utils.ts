export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'address',
    width: 95,
  },
];
