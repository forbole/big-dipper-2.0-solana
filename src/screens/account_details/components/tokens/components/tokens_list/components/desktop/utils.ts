export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 50,
  },
  {
    key: 'amount',
    align: 'right',
    width: 50,
  },
];
