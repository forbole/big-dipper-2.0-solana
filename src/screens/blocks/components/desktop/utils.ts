export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 10,
  },
  {
    key: 'leader',
    width: 35,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'txs',
    align: 'right',
    width: 10,
  },
  {
    key: 'time',
    align: 'right',
    width: 15,
  },
];
