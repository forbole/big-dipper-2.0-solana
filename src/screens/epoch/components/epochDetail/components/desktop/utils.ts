export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'name',
    width: 30,
  },
  {
    key: 'value',
    width: 20,
  },
  {
    key: 'detail',
    width: 50,
  },
];
