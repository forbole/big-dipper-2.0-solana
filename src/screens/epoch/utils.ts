import numeral from 'numeral';
import {
  InflationType,
  InflationGovernorType,
  EpochScheduleType,
} from './types';

export const inflationData = (data: InflationType, t: any) => {
  return ([
    {
      name: t('epoch'),
      value: data.epoch,
      detail: t('epochDetail'),
    },
    {
      name: t('totalInflation'),
      value: numeral(data.total).format('0.0[000]'),
      detail: t('totalInflationDetail'),
    },
    {
      name: t('validatorInflation'),
      value: numeral(data.validator).format('0.0[000]'),
      detail: t('validatorInflationDetail'),
    },
    {
      name: t('foundationInflation'),
      value: numeral(data.foundation).format('0.0[000]'),
      detail: t('foundationInflationDetail'),
    },
  ]);
};

export const inflationGovernorData = (data: InflationGovernorType, t: any) => {
  return ([
    {
      name: t('initial'),
      value: data.initial,
      detail: t('initialDetail'),
    },
    {
      name: t('terminal'),
      value: data.terminal,
      detail: t('terminalDetail'),
    },
    {
      name: t('taper'),
      value: data.taper,
      detail: t('taperDetail'),
    },
    {
      name: t('foundation'),
      value: data.foundation,
      detail: t('foundationDetail'),
    },
    {
      name: t('foundationTerm'),
      value: data.foundationTerm,
      detail: t('foundationTermDetail'),
    },
  ]);
};

export const epochScheduleData = (data: EpochScheduleType, t: any) => {
  return ([
    {
      name: t('slotsPerEpoch'),
      value: data.slotsPerEpoch,
      detail: t('slotsPerEpochDetail'),
    },
    {
      name: t('leaderScheduleSlotOffset'),
      value: data.leaderScheduleSlotOffset,
      detail: t('leaderScheduleSlotOffsetDetail'),
    },
    {
      name: t('warmup'),
      value: `${data.warmup}`.toUpperCase(),
      detail: t('warmupDetail'),
    },
    {
      name: t('firstNormalEpoch'),
      value: data.firstNormalEpoch,
      detail: t('firstNormalEpochDetail'),
    },
    {
      name: t('firstNormalSlot'),
      value: data.firstNormalSlot,
      detail: t('firstNormalSlotDetail'),
    },
  ]);
};
