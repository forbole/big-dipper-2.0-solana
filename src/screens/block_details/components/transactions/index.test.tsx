import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Transactions from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  TransactionsList: (props) => <div id="TransactionsList" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails/Transactions', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <Transactions
            transactions={[
              {
                slot: 123548722,
                signature: '4SGxuRMcseNbwki3tGxXPpfz7iFnuo9FUpTfiM4gJ8rhH59uZYSBBK2zW27xRdGX8Sb2N4VkGUnBYt59SBKEhPfB',
                success: true,
                timestamp: '2021-09-13T20:06:17.363145',
                numInstructions: 13,
              },
            ]}
          />
        </MockTheme>
      </RecoilRoot>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
