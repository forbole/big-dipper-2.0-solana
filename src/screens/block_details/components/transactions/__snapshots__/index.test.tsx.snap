// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails/Transactions matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <div
    className="makeStyles-header"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
    >
      transactions:transactions
    </h2>
  </div>
  <div
    className="makeStyles-list"
    hasNextPage={false}
    id="TransactionsList"
    isItemLoaded={[Function]}
    isNextPageLoading={false}
    itemCount={1}
    loadMoreItems={[Function]}
    loadNextPage={[Function]}
    transactions={
      Array [
        Object {
          "hash": "4SGxuRMcseNbwki3tGxXPpfz7iFnuo9FUpTfiM4gJ8rhH59uZYSBBK2zW27xRdGX8Sb2N4VkGUnBYt59SBKEhPfB",
          "numInstructions": 13,
          "slot": 123548722,
          "success": true,
          "timestamp": "2021-09-13T20:06:17.363145",
        },
      ]
    }
  />
</div>
`;
